src/Dapper.AOT.Analyzers/CodeAnalysis/DapperAnalyzer.cs

```diff
@@ -155,6 +160,39 @@ public void OnOperation(OperationAnalysisContext ctx)
                                 ValidatePropertyUsage(ctx, assignment.Value, false);
                             }
                         }
+                        break;
+                    case OperationKind.VariableDeclaration when ctx.Operation is IVariableDeclarationOperation variableDeclarationOperation
+                        && variableDeclarationOperation.Declarators.FirstOrDefault() is { } declarator
```

Comment:
Not sure we need to limit it to the first, not do in I think we need to limit it to variable declarators - even `DoThing(new SqlCommand("foo"))` should be targeted; can we catch the constructor syntax instead?

Reply:
changed to `ObjectCreation`